lib/thrift/generator/binary/framed/server.ex

```diff
@@ -99,15 +99,15 @@ defmodule Thrift.Generator.Binary.Framed.Server do
     quote do
       try do
         unquote(quoted_handler)
-      rescue
-        unquote(rescue_blocks)
       catch kind, reason ->
         formatted_exception = Exception.format(kind, reason, System.stacktrace)
         Logger.error("Exception not defined in thrift spec was thrown: #{formatted_exception}")
         error = Thrift.TApplicationException.exception(
           type: :internal_error,
           message: "Server error: #{formatted_exception}")
         {:server_error, error}
+      rescue
```

Comment:
Out of curiosity, should `rescue` always come after `catch` ? If so maybe we can add some static analysis and catch this at the erbert level

Reply:
Elixir 1.6 has a warning built in for this I believe

Reply:
@nichochar, yes, rescue should always come after catch (at least according to the new style rules). I don't think there's any technical reason for this, but 1.6 now emits a warning when this isn't the case.